### PR TITLE
implement sonar checks and apply fixes - #381

### PR DESCRIPTION
**Pull request by proddy**

### CODE REVIEW

**Comment by proddy** [comment]
I've integrated in sonarqube into the build process. It can be manually invoked via the CLI as in script/run_sonar.sh or on every pull in a new GitHub action.

File: src/helpers.cpp [hunk 1]
```diff
@@ -192,7 +192,7 @@ char * Helpers::render_boolean(char * result, bool value) {
 
 // render for native char strings
 char * Helpers::render_value(char * result, const char * value, const int8_t format __attribute__((unused))) {
-    strcpy(result, value);
+    strlcpy(result, value, sizeof(result));
```
Comment (MichaelDvP) [inline src/helpers.cpp:195]: the sizeof will always be a pointersize (4), not the stringlength.

Comment (proddy) [reply src/helpers.cpp:195]: oops, that's bad. I'll fix.

File: src/web/WebLogService.cpp [hunk 2]
```diff
@@ -158,7 +158,7 @@ void WebLogService::loop() {
 // convert time to real offset
 char * WebLogService::messagetime(char * out, const uint64_t t) {
     if (!time_offset_) {
-        strcpy(out, uuid::log::format_timestamp_ms(t, 3).c_str());
+        strlcpy(out, uuid::log::format_timestamp_ms(t, 3).c_str(), sizeof(out));
```
Comment (MichaelDvP) [inline src/web/WebLogService.cpp:161]: Same sizeof issue here, to check you'll need to pass the size to the messagetime-function

Comment (proddy) [reply src/web/WebLogService.cpp:161]: or use strlen(out) ?

Comment (MichaelDvP) [reply src/web/WebLogService.cpp:161]: No, declare as `char * WebLogService::messagetime(char * out, const uint64_t t, size_t  bufsize) {`, than you can use 
`strlcpy`. 
You have to determine the size where the busser is declared. `char buffer[20]` have sizeof(buffer) as 20, but`char * pbuffer=buffer` gives sizeof(pbuffer) with result 4. strlen is undefined if the buffer is not initialized.

Comment (proddy) [reply src/web/WebLogService.cpp:161]: fixed

**Comment by MichaelDvP** [comment]
looks good now, you also can declare the busffsize as const and replace the `25` in line 164/165  by `buffsize` to have all checks set to the right sizeof(time_string).

**Comment by proddy** [comment]
thanks for checking.

**Pull request closed without merging**